src/systems/systems/src/hydraulic/mod.rs

```diff
@@ -1622,6 +1625,185 @@ impl SimulationElement for LeakMeasurementValve {
     }
 }
 
+struct SpringPhysics {
+    k_constant: f64,
+    damping_constant: f64,
+
+    last_length: f64,
+}
+impl SpringPhysics {
+    fn new() -> Self {
+        Self {
+            k_constant: 5000.,
+            damping_constant: 500.,
+
+            last_length: 0.,
+        }
+    }
+
+    fn update_force_nm(
+        &mut self,
+        context: &UpdateContext,
+        position1: Vector3<f64>,
+        position2: Vector3<f64>,
+    ) -> Vector3<f64> {
+        let spring_vector = position1 - position2;
+        let spring_length = spring_vector.norm();
+
+        if spring_length.abs() < 0.001 {
+            self.last_length = spring_length;
+
+            return Vector3::default();
+        }
+
+        let spring_vector_normalized = spring_vector.normalize();
+        let velocity = (spring_length - self.last_length) / context.delta_as_secs_f64();
+
+        let k_force = spring_length * self.k_constant;
+        let damping_force = velocity * self.damping_constant;
+
+        self.last_length = spring_length;
+
+        (k_force + damping_force) * spring_vector_normalized
+    }
+}
+
+struct FluidPhysics {
+    reference_point_cg: Vector3<f64>,
+    fluid_cg_position: Vector3<f64>,
+    fluid_cg_speed: Vector3<f64>,
+
+    virtual_mass: Mass,
+    spring: SpringPhysics,
+    anisotropic_damping_constant: Vector3<f64>,
+
+    g_trap_is_empty: DelayedTrueLogicGate,
+}
+impl FluidPhysics {
+    const MEAN_G_TRAP_CAVITY_TIME_DURATION_SECONDS: f64 = 20.;
+    const STD_DEV_G_TRAP_CAVITY_TIME_DURATION_SECONDS: f64 = 4.;
+
+    fn new() -> Self {
+        Self {
+            reference_point_cg: Vector3::default(),
+            fluid_cg_position: Vector3::new(0., -0.2, 0.),
+            fluid_cg_speed: Vector3::default(),
+
+            virtual_mass: Mass::new::<kilogram>(100.),
+            spring: SpringPhysics::new(),
+            anisotropic_damping_constant: Vector3::new(25., 20., 25.),
```

Comment:
Do most of these constants not depend on the geometry of the container (e.g the reservoir)? If so, they should probably be passed in or maybe we could explicitly designate this struct to be `ReservoirFluidPhysics`

Reply:
This is really macro simulated and not using physical dimensions at all. So this is more like general constants that seem to work with fluid in reservoir in general, not dedicated to a A320 reservoir.

All the maps used however are what we'd change on a reservoir/plane basis I'd say.